core/src/main/java/com/linecorp/armeria/common/CommonPools.java

```diff
@@ -37,12 +35,8 @@
 
     static {
         // Threads spawned as needed and reused, with a 60s timeout and unbounded work queue.
-        final ScheduledThreadPoolExecutor scheduledThreadPoolExecutor = new ScheduledThreadPoolExecutor(
-                Flags.numCommonBlockingTaskThreads(),
-                ThreadFactories.newThreadFactory("armeria-common-blocking-tasks", true));
-        scheduledThreadPoolExecutor.setKeepAliveTime(60, TimeUnit.SECONDS);
-        scheduledThreadPoolExecutor.allowCoreThreadTimeOut(true);
-        BLOCKING_TASK_EXECUTOR = scheduledThreadPoolExecutor;
+        BLOCKING_TASK_EXECUTOR = BlockingTaskExecutorBuilder.builder()
+                                                            .build();
```

Comment:
`XXXBuilder.builder.build()` looks weird to me.
How about renaming to `BlockingTaskExecutors` and replacing this with `BlockingTaskExecutors.of()`?

Reply:
Perhaps we can add a builder to this method?
`CommonPools.blockingTaskExecutorBuilder()` or
`CommonPools.builderForBlockingTask()`?
Because we might want to add another builder for the worker group but we probably don't want to add another class for that builder.

Reply:
How about defining a new interface, like `BlockingTaskExecutor` that extends `ScheduledTaskExecutorService` and starting from there? We won't add much there but we may want to improve it over time, like modifying its methods to return a `CompletableFuture` (or `CompletionStage`).

Reply:
I think we need to consider this more.
Initial intention was to make variable `numThreads` when user wants to create blocking task executor given arbitrary number of threads.

And from @trustin comment, it would be beneficial to make another interface for future extension.
To create that interface, I think at least one feature is added into that interface.

What could be a good (maybe simple?) candidate for initial interface implementation?